src/utils/auth.ts

```diff
@@ -0,0 +1,30 @@
+import express, {Request, NextFunction} from 'express';
+const jwt = require('jsonwebtoken');
+import { PrismaClient } from '@prisma/client';
+
+const prisma = new PrismaClient();
+
+export interface ReqWithUser extends Request {
+    user: {}
+}
+
+// TODO: Test out this middleware once login endpoint is done and returns a JSON web token
+const authenticateJWT = async (req: ReqWithUser, next: NextFunction, prisma: PrismaClient) => {
+    const authHeader = req.headers.authorization;
+
+    if (authHeader) {
+        const token = authHeader.split(' ')[1];
+        try {
+            const user = await jwt.verify(token, process.env.ACCESS_TOKEN_SECRET)
+            req.user = await prisma.user.findUniqueOrThrow({
+                where: {
+                    id: user.id
+            }
```

Comment:
Make sure to delete the password - try not to be passing around the user's password all over your app. You should only need to look at the hashed password during login and register
```suggestion
            }
            delete req.user.password
```